File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/organisms/header/OrgNavigationHeaderActionItem.kt
package com.perrystreet.woof.designsystem.atomic.organisms.header

import androidx.annotation.DrawableRes
import androidx.compose.runtime.Immutable
import com.perrystreet.woof.designsystem.atomic.atoms.icon.roles.IconColorRole

@Immutable
data class OrgNavigationHeaderActionItem(
    @DrawableRes val iconRes: Int,
    val contentDescription: String,
    val colorRole: IconColorRole,
    val onTap: () -> Unit,
)

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Organisms/Header/OrgNavigationHeaderActionItem.swift
import Resources

public struct OrgNavigationHeaderActionItem {
    public let icon: ImageAsset
    public let contentDescription: String
    public let colorRole: IconColorRole
    public let onTap: () -> Void

    public init(icon: ImageAsset, contentDescription: String, colorRole: IconColorRole, onTap: @escaping () -> Void) {
        self.icon = icon
        self.contentDescription = contentDescription
        self.colorRole = colorRole
        self.onTap = onTap
    }
}
